salvo/src/lib/benchmark/test_binary_benchmark.py

```diff
@@ -0,0 +1,183 @@
+"""
+Test the fully dockerized benchmark class
+"""
+
+import pytest
+import subprocess
+from unittest.mock import patch
+
+import api.control_pb2 as proto_control
+
+from src.lib.benchmark import binary_benchmark
+from src.lib import generate_test_objects
+
+def test_no_sources():
+  """Test benchmark validation logic.
+
+  We expect the validation logic to throw an exception,
+  since no sources are present
+  """
+  # create a valid configuration with no images
+  job_control = generate_test_objects.generate_default_job_control()
+
+  generate_test_objects.generate_environment(job_control)
+
+  benchmark = binary_benchmark.Benchmark(job_control, "test_benchmark")
+
+  # Calling execute_benchmark should throw an exception
+  with pytest.raises(binary_benchmark.BinaryBenchmarkError) \
+      as validation_error:
+    benchmark.execute_benchmark()
+
+  assert str(validation_error.value) == "No source configuration specified"
+
+# Mock the build invocations so we don't actually try to build this big chungus
+@patch('src.lib.builder.nighthawk_builder.NightHawkBuilder.build_nighthawk_benchmarks')
```

Comment:
Can we make these a string constant above? The lines herein all tests  exceed 80 characters.